client/src/app/shared/components/sorting-list/sorting-list.component.ts

```diff
@@ -149,6 +149,14 @@ export class SortingListComponent implements OnInit, OnDestroy {
         }
     }
 
+    /**
+     * sorts the array by the new order
+     * @param newOrder the new order in the form of (id => position)
+     */
+    public updateOrder(newOrder: number[]): void {
```

Comment:
Please see the other comment for putting a store/restore mechanism in this component.